go/membufc/e2e/fields_extended_test.go

```diff
@@ -0,0 +1,126 @@
+		0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xa, 0xb, 0xc, 0xd, 0xe, 0xe, 0xf}
+}
+
+func getByte32Array() [][32]byte {
+	return [][32]byte{
+		{0x1, 0x1, 0x1, 0x1, 0x1, 0x1, 0x1, 0x1, 0x1, 0x1, 0x1, 0x1, 0x1, 0x1, 0x1, 0x1,
+			0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xa, 0xb, 0xc, 0xd, 0xe, 0xe, 0xf},
+		{0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xa, 0xb, 0xc, 0xd, 0xe, 0xe, 0xf,
+			0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xa, 0xb, 0xc, 0xd, 0xe, 0xe, 0xf},
+	}
+}
+
+func TestMembuffersFieldsExtended_HandlesFixedBytes32(t *testing.T) {
+	// prepare
+	data := getByte32Obj()
+
+	// build
+	msgBuilder := &types.WithFixedBytes32Builder{Foo: data}
+	msg := msgBuilder.Build()
+	t.Log(msg.String())
+
+	// check
+	require.EqualValues(t, data, msg.Foo())
+	//originalRaw := msg.Raw()
+	require.True(t, bytes.Equal(msg.Raw(), data[:]), "raw message should not include size")
+
+	// overwrite
+	msg2 := types.WithFixedBytes32BuilderFromRaw(msg.Raw()).Build()
+	require.EqualValues(t, data, msg2.Foo())
+	require.True(t, bytes.Equal(msg2.Raw(), data[:]), "raw message should be equal to the original value")
+
+	// mutate
+	other := [32]byte{0xfe, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab,
+		0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xab, 0xef}
+	require.NoError(t, msg.MutateFoo(other))
+	t.Log(msg.String())
+
+	// check mutate
+	require.EqualValues(t, other, msg.Foo())
+	require.True(t, bytes.Equal(msg.Raw(), other[:]), "raw message should be equal to the mutated value")
+
+	// check version with overwrite didn't change after mutate
+	require.EqualValues(t, data, msg2.Foo())
+
+	// check hexdump
+	require.NoError(t, msgBuilder.HexDump("msg ", 0))
+}
+
+func TestMembuffersFieldsExtended_HandlesFixedBytes32AndUint32(t *testing.T) {
+	bar := uint32(1977)
+	data := getByte32Obj()
+	msg := (&types.WithFixedBytes32AndUint32Builder{Foo: data, Bar: bar}).Build()
+
+	t.Log(msg.String())
+	require.EqualValues(t, data, msg.Foo())
+	require.EqualValues(t, bar, msg.Bar())
+
+	// check hexdump
+	require.NoError(t, (&types.WithFixedBytes32AndUint32Builder{Foo: data, Bar: bar}).HexDump("msg ", 0))
+}
+
+func TestMembuffersFieldsExtended_HandlesFixedBytes32InMiddle(t *testing.T) {
+	baz := uint32(1789)
+	bar := "1977"
+	data := getByte32Obj()
+	msg := (&types.WithFixedBytes32InMiddleBuilder{Baz: baz, Foo: data, Bar: bar}).Build()
+
+	t.Log(msg.String())
+	require.EqualValues(t, baz, msg.Baz())
+	require.EqualValues(t, data, msg.Foo())
+	require.EqualValues(t, bar, msg.Bar())
+}
+
+func TestMembuffersFieldsExtended_HandlesFixedBytes32Array(t *testing.T) {
+	data := getByte32Array()
+	msgBuilder := &types.WithRepeatedFixedBytes32Builder{Foo: data}
+	msg := msgBuilder.Build()
+
+	t.Log(msg.String())
+	t.Log(msg.Raw())
+	itr := msg.FooIterator()
+	oneArray := make([]byte, 4+len(data)*32)
```

Comment:
maybe instead of creating the expected raw, do the opposite, something like
```
	require.True(t, bytes.Equal(msg.Raw()[:31], data[0][:]))
	require.True(t, bytes.Equal(msg.Raw()[32:63], data[1][:]))
```

also note that you copy-pasted the failure message but it doesn't reflect the actual check